modules/apps/roles/roles-admin-web/src/main/resources/META-INF/resources/select_site_role.jspf

```diff
@@ -41,6 +42,14 @@ if (step == 1) {
 	}
 }
 
+LinkedHashMap<String, Object> groupParams = new LinkedHashMap<>();
+
+groupParams.put("inherit", Boolean.FALSE);
+groupParams.put("site", Boolean.TRUE);
+groupParams.put("usersGroups", selUser.getUserId());
+
+groups = GroupLocalServiceUtil.search(user.getCompanyId(), keywords, groupParams, QueryUtil.ALL_POS, QueryUtil.ALL_POS);
+
```

Comment:
This search looks fine to me, but putting it here means it will run no matter what, even if we don't care about searching for groups.  This view is step-based, and getting the user's groups is only relevant in the first step.  It looks to me like we need to make 2 changes here:

1. Branch on the presence of keywords inside the `if` block that checks the step above, executing the original group logic if there are no search terms
2. Make sure that the block with `filterManageableGroups` is called on the results of either branch.  In order to not introduce unexpected behavior when there is only one search result, we should copy that `if` block that checks `filterManageableGroups` into the search branch.